.github/workflows/pytest.yml

```diff
@@ -198,11 +198,24 @@ jobs:
           path: coverage
       - name: List downloaded files
         run: |
-          ls -R
+          find coverage -type f
       - name: Upload test coverage report to CodeCov
         uses: codecov/codecov-action@v4
         with:
           directory: coverage
+      - name: Install Micromamba
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          init-shell: bash
+          environment-name: coverage
+          create-args: >-
+            python=3.11
+            coverage>=7.4.1
+      - name: Combine coverage data and check that we have required coverage
+        run: |
+          micromamba run -n coverage coverage combine coverage/*/.coverage
+          micromamba run -n coverage coverage xml
+          micromamba run -n coverage coverage report --fail-under=100
```

Comment:
Seems like the simplest possible check -- just a fixed percentage threshold.

Could improve by using the `codecov` CLI to track whatever the ongoing coverage is, but this will at least ensure we don't backslide, and CodeCov will keep accumulating data.

Reply:
OK let's see if this thing fails! And then we can crank the # down.